### PR TITLE
Update testing instructions on CI doc

### PR DESCRIPTION
**Pull request by echeran**
Docs improvement. Prompted by discussion on PR #472, but it is more of a general task (not sure which issue # to assign).

### CODE REVIEW

File: docs/process/ci_build.md [hunk 1]
```diff
@@ -28,15 +28,24 @@ Basic concepts:
...
+
+So here are steps to test GH Actions change on your personal fork, with links to an example:
+
+1. create a new testing-only branch that tacks on [an extra commit to make testing-appropriate changes](https://github.com/echeran/icu4x/pull/22/commits/538176500f54594cffa1844c0244c7135ea24b84)
+2. push the testing branch to the personal fork of `icu4x` and create a [testing PR targeting personal fork's master](https://github.com/echeran/icu4x/pull/22)
+3. watch [GH Actions execute b/c of testing PR satisfying the execution trigger conditions](https://github.com/echeran/icu4x/actions/runs/539462550)
```
Comment (gregtatum) [inline docs/process/ci_build.md:45]: Suggestion: Replace `b/c` with `because` as it's clearer.

Suggestion: This phrase is a little awkward to interpret. Perhaps rephrase it to something like:

"GitHub Actions will run a new job because the testing PR satisfies the execution trigger conditions."

File: docs/process/ci_build.md [hunk 2]
```diff
@@ -28,15 +28,24 @@ Basic concepts:
...
+There might be reasons why you want or need to test changes to GH Actions on your personal fork. Less frequently, there might be specific changes that cannot be tested via a PR because they only happen on merges to the main branch `master`. For example, API docs changes and benchmark dashboard changes should only occur on merges to master. In this case, you can use your personal fork of the upstream repo as a testing ground. The naive approach is not recommended -- to directly modify your `master` branch -- because it requires awkwardly changing your git repo during and after testing in ways that are often confusing and opposite to "git-flow" habits. 
+
+So here are steps to test GH Actions change on your personal fork, with links to an example:
+
+1. create a new testing-only branch that tacks on [an extra commit to make testing-appropriate changes](https://github.com/echeran/icu4x/pull/22/commits/538176500f54594cffa1844c0244c7135ea24b84)
```
Comment (gregtatum) [inline docs/process/ci_build.md:43]: Suggestion: I generally prefer complete sentences with documentation, with proper capitalization and punctuation as it's easier to read an understand.

File: docs/process/ci_build.md [hunk 3]
```diff
@@ -28,15 +28,24 @@ Basic concepts:
...
-  * For this simplistic example, the way to improve it to allow a Pull Request to run the same checks after each new commit is pushed, in the same way that checks are run on merges to `master`, is to change `on.pull_request.branches` to [match all origin branch names](https://github.com/unicode-org/icu4x/commit/16ae4611738fbe94b36e17b77aee6cc541c0a171).
+  * For this simplistic example, the way to improve it in order to allow a Pull Request to run the same checks after each new commit is pushed, in the same way that checks are run on merges to `master`, is to change `on.pull_request.branches` to [match all origin branch names](https://github.com/unicode-org/icu4x/commit/16ae4611738fbe94b36e17b77aee6cc541c0a171).
 
 ## Testing a Workflow
```
Comment (gregtatum) [inline docs/process/ci_build.md:33]: Suggestion: I would split this out into two different sections to make it easier to scan.

`## Testing a Workflow in unicode-org/icu4x`

And

`## Testing a Workflow in Your Forked Repo`

File: docs/process/ci_build.md [hunk 4]
```diff
@@ -102,33 +111,41 @@ Less frequently, there might be specific changes that cannot be tested via a PR
   * There is no mechanism to persist data storage across workflow instantiations. The only way to persist / store data across workflow instances is through making commits on a branch within the git repo itself. A benchmark dashboard requires the accumulation of data points computed from each invocation. Therefore, the benchmark action that we use relies on creating git commits on a branch to store this historical information over time.
 
 
-## Implementation of Various CI Functionality
```
Comment (gregtatum) [inline docs/process/ci_build.md:105]: Praise: Thanks, for the clarification here.

File: docs/process/ci_build.md [hunk 5]
```diff
@@ -102,33 +111,41 @@ Less frequently, there might be specific changes that cannot be tested via a PR
...
-      - [config for Codecov](https://github.com/unicode-org/icu4x/commit/fd524a91df3f5aef2efb908adddd051827972328#diff-a2115d277b5ca5a2f09a999e53440839cf332b94da177f3d1766334555b0f7c6R49-R53)
-    * [Coveralls](https://coveralls.io/)
-      - [config for Coveralls](https://github.com/unicode-org/icu4x/commit/00c514f3a189787c4e77704eb98714e6b6430d37#diff-a2115d277b5ca5a2f09a999e53440839cf332b94da177f3d1766334555b0f7c6R36-R40)
-  - Both code coverage dashboard services automatically support most output file formats for most programming languages' code coverage tools
\ No newline at end of file
+## Examples of Various CI Functionality
+
+### Copy API docs to separate repo
+
+- We maintain a separate repo to store API docs. This allows the generated output HTML, which is usually larger in size than the source code it is generate from, to be under version control without swelling up the size of the main repo. Also, changes to the APIs will cause larger diffs in the HTML.
```
Comment (gregtatum) [inline docs/process/ci_build.md:118]: Suggestion: I find bullet point lists harder to read than paragraph form. Most of these bullets are only delineating paragraphs or sentences. I believe all of this copy reads fine in paragraph form, and only the sub-lists should be retained.

File: docs/process/ci_build.md [hunk 6]
```diff
@@ -102,33 +111,41 @@ Less frequently, there might be specific changes that cannot be tested via a PR
...
-  - Code coverage is configured in a separate workflow since it is completely independent of the operations for build, test, format/style-checking, linting/error-prone-checking, API docs, benchmarking.
-  - Two code coverage dashboard services exist:
-    * [Codecov](https://codecov.io/)
-      - [config for Codecov](https://github.com/unicode-org/icu4x/commit/fd524a91df3f5aef2efb908adddd051827972328#diff-a2115d277b5ca5a2f09a999e53440839cf332b94da177f3d1766334555b0f7c6R49-R53)
-    * [Coveralls](https://coveralls.io/)
-      - [config for Coveralls](https://github.com/unicode-org/icu4x/commit/00c514f3a189787c4e77704eb98714e6b6430d37#diff-a2115d277b5ca5a2f09a999e53440839cf332b94da177f3d1766334555b0f7c6R36-R40)
-  - Both code coverage dashboard services automatically support most output file formats for most programming languages' code coverage tools
\ No newline at end of file
+## Examples of Various CI Functionality
+
```
Comment (gregtatum) [inline docs/process/ci_build.md:115]: Suggestion: I would explicitly call out that the `.yml` files should provide detailed documentation for the individual steps, and that these are only broad examples of what the GitHub Actions can do.

File: docs/process/ci_build.md [hunk 7]
```diff
@@ -28,15 +28,24 @@ Basic concepts:
...
+There might be reasons why you want or need to test changes to GH Actions on your personal fork. Less frequently, there might be specific changes that cannot be tested via a PR because they only happen on merges to the main branch `master`. For example, API docs changes and benchmark dashboard changes should only occur on merges to master. In this case, you can use your personal fork of the upstream repo as a testing ground. The naive approach is not recommended -- to directly modify your `master` branch -- because it requires awkwardly changing your git repo during and after testing in ways that are often confusing and opposite to "git-flow" habits.
```
Comment (gregtatum) [inline docs/process/ci_build.md:39]: Suggestion: I think abbreviations here make the wording harder to interpret as a new reader. I would prefer to use the full name of "GitHub Actions" everywhere.

**Review by gregtatum — APPROVED** [review]
I left a few suggestions, mostly on style and readability, rather than correctness. I'm marking as approved, but feel free to flag me for review again if you want me to double-check any additional changes.

File: docs/process/ci_build.md (hunk 6 above)
Comment (echeran) [reply docs/process/ci_build.md:117]: Done.

File: docs/process/ci_build.md (hunk 1 above)
Comment (echeran) [reply docs/process/ci_build.md]: Done.

File: docs/process/ci_build.md (hunk 2 above)
Comment (echeran) [reply docs/process/ci_build.md]: Done. Thanks for the suggestion. I hadn't thought about that, but it makes sense, I'll keep that in mind for the future.

File: docs/process/ci_build.md (hunk 3 above)
Comment (echeran) [reply docs/process/ci_build.md]: Done.

File: docs/process/ci_build.md (hunk 5 above)
Comment (echeran) [reply docs/process/ci_build.md]: Done.

File: docs/process/ci_build.md (hunk 6 above)
Comment (echeran) [reply docs/process/ci_build.md:117]: Done.

File: docs/process/ci_build.md (hunk 7 above)
Comment (echeran) [reply docs/process/ci_build.md]: Done.

**Review by gregtatum — APPROVED** [review]
Thanks for the changes! Looks good to me.
